packages/js/src/generators/release-version/release-version.ts

```diff
@@ -583,3 +757,58 @@ function getColor(projectName: string) {
 
   return colors[colorIndex];
 }
+
+function sortProjectsTopologically(
```

Comment:
Ensure that this handles circular dependencies. It would be a good unit test to have.

Reply:
Circular dependencies in versioning are handled and unit tested in: https://github.com/nrwl/nx/pull/23252/commits/ee11e208363fb641f69843f274e973c7d67d11ec